project.clj

```diff
@@ -1,4 +1,4 @@
-(defproject lupapiste/commons "2.0.4"
+(defproject lupapiste/commons "3.0.0"
```

Comment:
Vähän makuasia, mutta muutos ei nähdäkseni ole ihan niin rikkova, että pitäisi pääversiota korottaa. Eihän siitä toiki haittaakaan ole.

Reply:
Kun loppujen lopuksihan muutos kuitenkin on pääosin alaspäin yhteensopiva? Eli vaikka liitetyypit ryhmitellään eri tavalla, niin varsinaiset tyypit kuitenkin säilyvät, eikö?

Reply:
Muistelen jostain lukeneeni että semanttista versiointia olisi tarkoitus harrastaa ja siinä olen itse seurannut tätä ohjeistusta: https://semver.org/  Muistinko väärin? Ja jos muistin, niin löytyykö jostain ohjeistusta versiointiin, jotta osaan tehdä jatkossa oikein? Joskus aiemminkin huomautit tästä, ja haluaisin osata tehdä tällaiset perusjutut oikein ja talon käytäntöjen mukaisesti, niin ettei tartte olla toistuvasti huomauttamassa.

Tässä nimenomanaisessa tapauksessa muutos on selvästi rikkova, eli kirjaston käyttäjän pitää tietyissä tapauksessa tehdä muutoksia koodiin. Itse kyllä diggailen siitä, että semanttinen versiointi tekee tästä ilmeisempää. Lupapiste esim. rasahtaa 100% varmasti ellei tee muutoksia koodiin ja nyt jää versio historiaan selkeä vinkki mahdollisista ongelman syistä. Eli jos tähtäin on kirjastojen semanttisessa versioinnissa, niin ei ole versiointi tässä makuasia sikäli että muutos on ilmeisen rikkova, mutta yleisemmällä tasolla versiointi sitä toki on ja lisäksi myös käytännöllisyys asia. Hyvä olisi olla konsistentti, jotta versionnista on maksimaalinen hyöty.

Reply:
Varsinaiset tyypit toki säilyvät ja esim. lupapisteen loppukäyttäjälle muutos ei näy rikkovana, mutta kirjaston käyttäjän (eli tässä tapauksessa lupapistettä tai onkaloa koodaavan näkökulmasta) tulee muutoksia joka vaatii muutoksia sellaisiin osiin softaa jotka ovat aiemmalla versiolla toimineet, ja sikäli kirjaston pääversion päivitys on mielestäni perusteltu.